lib/Backend/LowerMDShared.cpp

```diff
@@ -7711,12 +7711,24 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
     switch (instr->m_opcode)
     {
     case Js::OpCode::InlineMathSqrt:
-        // Sqrt maps directly to the SSE2 instruction.
-        // src and dst should already be XMM registers, all we need is just change the opcode.
-        Assert(helperMethod == (IR::JnHelperMethod)0);
-        Assert(instr->GetSrc2() == nullptr);
-        instr->m_opcode = instr->GetSrc1()->IsFloat64() ? Js::OpCode::SQRTSD : Js::OpCode::SQRTSS;
-        break;
+		{
+			// Sqrt maps directly to the SSE2 instruction.
+			// src and dst should already be XMM registers, all we need is just change the opcode.
+			Assert(helperMethod == (IR::JnHelperMethod)0);
+			Assert(instr->GetSrc2() == nullptr);
+			instr->m_opcode = instr->GetSrc1()->IsFloat64() ? Js::OpCode::SQRTSD : Js::OpCode::SQRTSS;
+
+			IR::Opnd *src = instr->GetSrc1();
+			IR::Opnd *dst = instr->GetDst();
+			if (!src->IsEqual(dst))
+			{
+				// Force source to be the same as destination to break false dependency on the register
+				Lowerer::InsertMove(dst, src, instr, true /* generateWriteBarrier */);
```

Comment:
>InsertMove [](start = 13, length = 10)

Could add an assert that both dst/src are regopnd, in which case generateWriteBarrier can be false.